### PR TITLE
Vmware guest

### PR DESCRIPTION
**Pull request by dav1x**
Added vmware guest module to ansible and removed a bunch of unneeded roles and variables.

Added INI support and added in 2 option netmask and v datacenter.

Tested this branch from scratch and works fine.

### CODE REVIEW

File: reference-architecture/vmware-ansible/README.md [hunk 1]
```diff
@@ -19,18 +19,19 @@ rpm -Uvh https://dl.fedoraproject.org/pub/epel/epel-release-latest-7.noarch.rpm*
...
-                  python-httplib2
-git clone https://github.com/dannvix/pySphere && cd pySphere/ && python setup.py install
+                  python-simplejson \
+                  python-six \
+
 git clone https://github.com/vmware/pyvmomi && cd pyvmomi/ && python setup.py install
 
 # Grabbed the patched vsphere_guest until the PR is merged
-cp vsphere_guest.py /usr/lib/python2.7/site-packages/ansible/modules/core/cloud/vmware/
+cp vmware_guest.py /usr/lib/python2.7/site-packages/ansible/modules/core/cloud/vmware/
```
Comment (cooktheryan) [inline reference-architecture/vmware-ansible/README.md]: Davis can you see if you can use the module similar to this?

https://github.com/openshift/openshift-ansible-contrib/tree/master/reference-architecture/aws-ansible/playbooks/library

**Comment by dav1x** [comment]
@cooktheryan you mean to just store the module for use?

File: reference-architecture/vmware-ansible/README.md (hunk 1 above)
Comment (cooktheryan) [inline reference-architecture/vmware-ansible/README.md]: From the comment below.

Yea just store it within the repo and don't copy it to the ansible directory. One concern I would have would be overwriting something in the current Ansible directory or if an Ansible patch comes out the module may be overwrote or deleted and that could break us. If we ship it within our repo then we know we have a known good module until upstream makes it to fully supported.

File: roles/validate-etcd/tasks/main.yaml [hunk 2]
```diff
@@ -5,7 +5,7 @@
   register: etcd_result
 
 - name: Validate etcd
-  command: "etcdctl -C https://{{ ansible_nodename }}:2379 --ca-file=/etc/origin/master/master.etcd-ca.crt --cert-file=/etc/origin/master/master.etcd-client.crt --key-file=/etc/origin/master/master.etcd-client.key cluster-health | grep 'cluster is'"
+  command: "etcdctl -C https://{{ ansible_fqdn }}:2379 --ca-file=/etc/origin/master/master.etcd-ca.crt --cert-file=/etc/origin/master/master.etcd-client.crt --key-file=/etc/origin/master/master.etcd-client.key cluster-health | grep 'cluster is'"
```
Comment (cooktheryan) [inline roles/validate-etcd/tasks/main.yaml]: This works perfectly.

File: reference-architecture/vmware-ansible/README.md (hunk 1 above)
Comment (dav1x) [inline reference-architecture/vmware-ansible/README.md]: The only problem is I can't call the module as you would in a normal playbook if I dont move it to that directory. 

I could rename it as to not overwrite the existing vmware_guest. vmware_guest_ocp or something like that

Comment (dav1x) [inline reference-architecture/vmware-ansible/README.md]: done!

**Pull request merged**